src/Symfony/Component/Form/Console/Descriptor/Descriptor.php

```
@@ -0,0 +1,144 @@
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ *
+ * @internal
+ */
+abstract class Descriptor implements DescriptorInterface
+{
+    /**
+     * @var SymfonyStyle
+     */
+    protected $output;
+    protected $type;
+    protected $ownOptions = array();
+    protected $overriddenOptions = array();
+    protected $inheritedOptions = array();
+    protected $extensionOptions = array();
+    protected $requiredOptions = array();
+    protected $parents = array();
+    protected $extensions = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function describe(OutputInterface $output, $object, array $options = array())
+    {
+        $this->output = $output;
+
+        switch (true) {
+            case $object instanceof ResolvedFormTypeInterface:
+                $this->describeResolvedFormType($object, $options);
+                break;
+        }
+    }
+
+    abstract protected function describeResolvedFormType(ResolvedFormTypeInterface $resolvedFormType, array $options = array());
+
+    public function isRequiredOption($option)
+    {
+        return in_array($option, $this->requiredOptions, true);
+    }
+
+    public function hasParents()
+    {
+        return 0 !== count($this->parents);
+    }
+
+    public function getParents()
+    {
+        return array_keys($this->parents);
+    }
+
+    public function hasExtensions()
+    {
+        return 0 !== count($this->extensions);
+    }
+
+    public function getExtensions()
+    {
+        return array_keys($this->extensions);
+    }
+
+    public function collectOptions(ResolvedFormTypeInterface $type)
+    {
+        $this->parents = array();
+        $this->extensions = array();
+
+        $ownOptionsResolver = new OptionsResolver();
+        try {
+            $type->getInnerType()->configureOptions($ownOptionsResolver);
+        } catch (UndefinedOptionsException $e) {
```

I see 🤔. It'll produce this issue whenever a form type tries to call `OptionResolver::addAllowedTypes/addAllowedValues/setAllowedTypes/setAllowedValues/setNormalizer()` on an option defined by one of the parent form type or extension.
As I think such cases won't happen often (appart `setNormalizer` to me), I think this is acceptable and I don't see any other solution right now.

But we definitively can't just silently catch the exception and display data already collected about the type without leaving a note to the user (not all options might have been properly collected).